### PR TITLE
fix(UIForms/TUI-197): nested array items key has now the indexes

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
Array inject the item index only on the first level of schema items.
For more complex item widgets, the key is not propagated.

Example: 
ArrayWidget
.    .CustomFieldset
.    .    .TexWidget // this has not the index in the key

On submit, the value is not passed.

**What is the chosen solution to this problem?**
Resolve keys deeply on array item schema computation

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/UIForm/fieldsets/Array/Array.component.js [hunk 1]
```diff
@@ -6,12 +6,16 @@ import defaultTemplates from '../../utils/templates';
 import defaultWidgets from '../../utils/widgets';
 
 function adaptKeyWithIndex(keys, index) {
-	let indexedKeys = keys;
-	const firstIndexPlaceholder = indexedKeys.indexOf('');
-	if (firstIndexPlaceholder >= 0) {
-		indexedKeys = [...keys];
-		indexedKeys[firstIndexPlaceholder] = index;
+	if (!keys) {
```
Comment (sgangnant) [inline packages/forms/src/UIForm/fieldsets/Array/Array.component.js]: This check could be made by callers before calling adaptKeyWithIndex to have a consistent return type from this method

File: packages/forms/src/UIForm/fieldsets/Array/Array.component.js [hunk 2]
```diff
@@ -6,12 +6,16 @@ import defaultTemplates from '../../utils/templates';
...
+		return;
 	}
+
+	let firstIndexPlaceholder = keys.indexOf('');
```
Comment (acateland) [inline packages/forms/src/UIForm/fieldsets/Array/Array.component.js]: ok i don't understand why you need to add this bit of code

Comment (jsomsanith) [reply packages/forms/src/UIForm/fieldsets/Array/Array.component.js]: The cases : 
- key = ["my", "array", "", "nested"] for nested items fields
- key = ["my", "array"] : this defines array itself. Each item will receive a key ["my", "array", index]
To check that, I spot the first empty string in the key
- find it:  and replace it, it's a nested element
- not found : it's an item key, I add the index after

File: packages/forms/src/UIForm/fieldsets/Array/Array.component.js (hunk 1 above)
Comment (jsomsanith) [reply packages/forms/src/UIForm/fieldsets/Array/Array.component.js]: Thanks, fixed

**Pull request merged**